provider/win_utils.py

```diff
@@ -0,0 +1,166 @@
+    test.log.info("Try to %s VirtioFsSvc service." % action)
+    session.cmd(cmd)
+    output = session.cmd_output(viofs_sc_query_cmd)
+    if expect_status not in output:
+        test.fail("Could not %s VirtioFsSvc service, "
+                  "detail: '%s'" % (action, output))
+
+
+@error_context.context_aware
+def viofs_sc_run(test, params, session):
+    viofs_sc_create_cmd = params["viofs_sc_create_cmd"]
+    viofs_sc_start_cmd = params["viofs_sc_start_cmd"]
+    viofs_sc_query_cmd = params["viofs_sc_query_cmd"]
+    viofs_sc_delete_cmd = params["viofs_sc_delete_cmd"]
+
+    output = session.cmd_output(viofs_sc_query_cmd)
+    if "not exist as an installed service" not in output:
+        error_context.context("Clean virtiofs service registered")
+        session.cmd(viofs_sc_delete_cmd)
+    test.log.info("Register virtiofs service")
+    viofs_svc_create(test, params, session, viofs_sc_create_cmd)
+
+    test.log.info("Check if virtiofs service is started.")
+    status, output = session.cmd_status_output(viofs_sc_query_cmd)
+    if "RUNNING" not in output:
+        test.log.info("Start virtiofs service.")
+        viofs_svc_stop_start(test, params, session, "start",
+                             viofs_sc_start_cmd, "RUNNING")
+    else:
+        test.log.info("Virtiofs service is running.")
+
+
+def install_winfsp(test, params, session):
+    cmd_timeout = params.get_numeric("cmd_timeout", 120)
+    install_path = params["install_path"]
+    check_installed_cmd = params["check_installed_cmd"] % install_path
+    # install winfsp tool
+    test.log.info("Install winfsp for windows guest.")
+    installed = session.cmd_status(check_installed_cmd) == 0
+    if installed:
+        test.log.info("Winfsp tool is already installed.")
+    else:
+        install_cmd = utils_misc.set_winutils_letter(session,
+                                                     params["install_cmd"])
+        session.cmd(install_cmd, cmd_timeout)
+        if not utils_misc.wait_for(lambda: not session.cmd_status(
+                check_installed_cmd), 60):
+            test.error("Winfsp tool is not installed.")
+
+
+def get_fs_dest_from_vm(test, params, session):
+    test_file = params.get('test_file', "test_file")
+    virtio_fs_disk_label = params.get("fs_target")
+    test.log.info("Get Volume letter of virtio fs target, the disk lable is "
+                  "%s." % virtio_fs_disk_label)
+    vol_con = "VolumeName='%s'" % virtio_fs_disk_label
+    volume_letter = utils_misc.wait_for(
+        lambda: utils_misc.get_win_disk_vol(session, condition=vol_con),
+        timeout=120)
+    if volume_letter is None:
+        test.fail("Could not get virtio-fs mounted volume letter.")
+    fs_dest = "%s:" % volume_letter
+
+    guest_file = os.path.join(fs_dest, test_file)
+    test.log.info("The guest file in shared dir is %s", guest_file)
+    return guest_file, fs_dest, volume_letter
+
+
+def basic_io_test(test, params, session):
```

Comment:
please check if qemu_img_utils.save_random_file_to_vm and qemu_img_utils.check_md5sum work for you (provider/qemu_img_utils.py)

Reply:
> please check if qemu_img_utils.save_random_file_to_vm and qemu_img_utils.check_md5sum work for you (provider/qemu_img_utils.py)

Hello zhenchao,  
After I've explored the codes for both qemu_img_utils.save_random_file_to_vm and qemu_img_utils.check_md5sum, I believe they are NOT adapted the context in this case.
qemu_img_utils.save_random_file_to_vm: The file name is generated automatically and is NOT returned, therefore, I can't get the file name for next step "md5 check"
qemu_img_utils.check_md5sum: This is designed for Linux platform only. However, this case must be run at Windows platform.
Thanks!